src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentDataServiceImpl.java

```diff
@@ -30,167 +27,111 @@
-        getJsonFromCloud(caseId, caseBucket, "Could not retrieve CollectionCase object from cloud");
-    if (collectionCaseStrOpt.isPresent()) {
-      return deserialiseCollectionCase(caseId, collectionCaseStrOpt.get());
-    } else {
-      return Optional.empty();
-    }
+    return cloudDataStore.retrieveObject(CollectionCase.class, caseSchema, caseId);
   }
 
-  private void writeJsonToCloud(String jsonInString, String key, String bucket, String debugMessage)
-      throws CTPException {
-    try {
-      cloudDataStore.storeObject(bucket, key, jsonInString);
-    } catch (StorageException e) {
-      log.with(key).error("Could not store json object in cloud", e);
-      throw new CTPException(Fault.SYSTEM_ERROR);
-    }
-    log.with(key).debug(debugMessage);
-  }
-
-  private Optional<UAC> deserialiseUAC(String universalAccessCode, String uacStrOpt)
-      throws CTPException {
-    Optional<UAC> uacOpt = Optional.empty();
-    if (uacStrOpt != null) {
-      ObjectMapper mapper = new ObjectMapper();
-      try {
-        uacOpt = Optional.of(mapper.readValue(uacStrOpt, UAC.class));
-      } catch (IOException e) {
-        log.with(universalAccessCode).error("Could not de-serialise UAC object from JSON", e);
-        throw new CTPException(Fault.SYSTEM_ERROR, e);
-      }
-      log.with(universalAccessCode).debug("UAC object has been retrieved from the cloud");
-    }
-    return uacOpt;
-  }
-
-  private Optional<CollectionCase> deserialiseCollectionCase(
-      String caseId, String collectionCaseStrOpt) throws CTPException {
-    Optional<CollectionCase> collectionCaseOpt = Optional.empty();
-    if (collectionCaseStrOpt != null) {
-      ObjectMapper mapper = new ObjectMapper();
-      try {
-        collectionCaseOpt =
-            Optional.of(mapper.readValue(collectionCaseStrOpt, CollectionCase.class));
-      } catch (IOException e) {
-        log.with(caseId).error("Could not de-serialise CollectionCase object from JSON", e);
-        throw new CTPException(Fault.SYSTEM_ERROR, e);
-      }
-      log.with(caseId).debug("CollectionCase object has been retrieved from the cloud");
+  /**
+   * Read a Case object from cloud based on its uprn.
+   *
+   * @param uprn - is the uprn that the target case must contain.
+   * @return - de-serialised version of the stored object
+   * @throws CTPException - if a cloud exception was detected.
+   */
+  @Override
+  public Optional<CollectionCase> readCollectionCaseByUprn(final String uprn) throws CTPException {
+    // Run search
+    String[] searchByUprnPath = new String[] {"address", "uprn"};
+    List<CollectionCase> searchResults =
+        cloudDataStore.search(CollectionCase.class, caseSchema, searchByUprnPath, uprn);
+
+    Optional<CollectionCase> collectionCase;
+    if (searchResults.isEmpty()) {
+      collectionCase = Optional.empty();
+    } else if (searchResults.size() == 1) {
+      collectionCase = Optional.of(searchResults.get(0));
+    } else {
```

Comment:
Will be a valid scenario to have more than one case for a upon (Household and Individuals at same address, flat 1a and flat flat 1b etc)

Reply:
Method now returns a list of matching cases.